### ISSUE TITLE
Client (5.1/5.2) repeatedly syncs the same target whilst others wait indefinitely on "waiting to sync".

### ISSUE THREAD

**Issue by Synthlight** [owncloud/client#11382]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

Ever since upgrading to 5.0 I've had issues with either the sync not starting for some items, or the sync stuck on "waiting to sync".

In 5.0 it just synced about 20% of the items (40 sync entries) and then just stopped. The rest remained on "waiting to sync" and never progressed. I'd have to repeatedly pause/unpause and "force sync now" multiple times to get it to finally sync the folder. (And that would be a one time thing I'd have to repeat to sync one entry.)

So I saw an issue (https://github.com/owncloud/client/issues/11308) that was resolved in 5.1 so after waiting for that release and trying it..... now it just syncs the same ones that were working before over and over whilst ignoring the rest.

I've tried re-adding the account and while only adding about 3 of 40 items I'd have synced, and the result is unchanged.

Here's a video I took of it doing the same one over and over (and it'll do this for hours) whilst never syncing the other entry.
This is with about 40 sync entries in the list:
https://github.com/owncloud/client/assets/1075166/203210bb-7c25-4048-8091-5793a853a51e

And here's after removing and adding the account again, but this time with only 3 items configured.
It syncs all of them, but keeps syncing over and over in a seemingly random order.
https://github.com/owncloud/client/assets/1075166/0499a177-dfb3-48dd-9994-01f66cc4694b
Here's a log from running this with the three:
[ownCloud.log](https://github.com/owncloud/client/files/13404324/ownCloud.log)

This *sort of* works with three folders since it eventually get to the others, but the more I add, the more ignored some get and the more repeated a handful of them become.
But on startup, this still has the issue of repeatedly syncing one of the entries many times before eventually getting to the other two. The more sync entries I have, the less likely one is to eventually be reached.

This is targeting my own local server which is currently running 10.13.2.3.
Current client version is 5.1.2.12600 on win 10 21H2 (LTSC).
This same issue happens on my ubuntu 22.04 clients. 4.2 works perfectly, 5.0+ has the same issue on whatever platform I try.

### Expected behavior

I expect it to sync with a similar frequency as in 4.2. Sync all folders in sequence on startup and then wait for changes.

### Steps to reproduce the issue

1: Add multiple sync entries. (40 in my case.)
2: Restart the client.
3: Watch some sync over and over whilst others wait indefinitely, sometimes eventually syncing hours later.

### Screenshots

Here's a video I took of it doing the same one over and over (and it'll do this for hours) whilst never syncing the other entry.
This is with about 40 sync entries in the list:
https://github.com/owncloud/client/assets/1075166/203210bb-7c25-4048-8091-5793a853a51e

And here's after removing and adding the account again, but this time with only 3 items configured.
It syncs all of them, but keeps syncing over and over in a seemingly random order.
https://github.com/owncloud/client/assets/1075166/0499a177-dfb3-48dd-9994-01f66cc4694b
Here's a log from running this with the three:
[ownCloud.log](https://github.com/owncloud/client/files/13404375/ownCloud.log)

### Logs

Here's a log from running this with the three:
[ownCloud.log](https://github.com/owncloud/client/files/13404375/ownCloud.log)

### Client version number

This is targeting my own local server which is currently running 10.13.2.3.
Current client version is 5.1.2.12600 on win 10 21H2 (LTSC).
This same issue happens on my ubuntu 22.04 clients. 4.2 works perfectly, 5.0+ has the same issue on whatever platform I try.

### Desktop environment (Linux only)

KDE

### Client package version and origin (Linux only)

IIRC it's `https://download.owncloud.com/desktop/ownCloud/stable/latest/linux/Ubuntu_22.04/`

### Installation path (Windows only)

`C:\Program Files (x86)\ownCloud\`

### Server information

This is targeting my own local server which is currently running 10.13.2.3. (Deb 11)

PHP 7.4.33 (cli) (built: Sep  4 2023 08:11:54) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.33, Copyright (c), by Zend Technologies

**Comment by tack0974** [comment]
having the exact same issue on various systems (Linux / MacOS), tried setting 'pollinterval' on the server side, but no luck

**Comment by bialy77** [comment]
Same here. Windows client. 

Preview how it looks from user side: https://streamable.com/35xqpw

**Comment by sneerix** [comment]
Same here after update to 5.1.2.12600 on Linux Ubuntu 22.04
Few minutes before on 4.x it was working perfectly fine.

**Comment by TheOneRing** [comment]
@Synthlight could you please run with the http logger? I'm interested in the response to https://server.redacted.url/owncloud/owncloud/app/list

It looks like the request to https://server.redacted.url/owncloud/owncloud/app/list fails and this triggers the account to revalidate its connection state. Possibly sending it into a loop.

**Comment by sneerix** [comment]
yes, it looks like #11296, 
GET /**owncloud/owncloud**/app/list HTTP/1.1" 302

**Comment by tack0974** [comment]
@TheOneRing  thanks, that fixed the issue (at least for me)

**Comment by Synthlight** [comment]
> @Synthlight could you please run with the http logger? I'm interested in the response to https://server.redacted.url/owncloud/owncloud/app/list
> 
> It looks like the request to https://server.redacted.url/owncloud/owncloud/app/list fails and this triggers the account to revalidate its connection state. Possibly sending it into a loop.

It gets a 302/redirect.

I also see `Unsupported redirect on "https://server.redacted.url/owncloud/owncloud/app/list" to "https://server.redacted.url/owncloud/apps/files/"` in the log.

```
23-11-21 19:53:20:136 [ info sync.httplogger ]:	"2191fb4e-d3a2-476a-9b60-0b536590c3c4: Response: GET 302 (Piplined,405ms) https://server.redacted.url/owncloud/owncloud/app/list Header: { Date: Wed, 22 Nov 2023 00:53:19 GMT, Server: Apache/2.4.56 (Debian), Strict-Transport-Security: max-age=15552000; includeSubDomains, X-Content-Type-Options: nosniff, X-XSS-Protection: 0, X-Robots-Tag: none, X-Frame-Options: SAMEORIGIN, X-Download-Options: noopen, X-Permitted-Cross-Domain-Policies: none, Expires: Thu, 19 Nov 1981 08:52:00 GMT, Cache-Control: no-store, no-cache, must-revalidate, Pragma: no-cache, Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *, Location: https://server.redacted.url/owncloud/apps/files/, Content-Length: 0, Content-Type: text/html; charset=UTF-8, Set-Cookie: oc_sessionPassphrase=__redacted__; expires=Wed, 22-Nov-2023 01:13:19 GMT; Max-Age=1200; path=/owncloud; secure; HttpOnly; SameSite=Strict, Keep-Alive: timeout=5, max=97, Connection: Keep-Alive, } Data: []"
23-11-21 19:53:20:136 [ warning gui.account.state ]:	checkConnectivity blocking: true
23-11-21 19:53:20:136 [ debug sync.networkjob.jobqueue ]	[ OCC::JobQueue::block ]:	block: 1 "myName@server.redacted.url"
23-11-21 19:53:20:136 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::checkServer ]:	Checking server and authentication
23-11-21 19:53:20:136 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::checkServer ]:	Trying to look up system proxy
23-11-21 19:53:20:136 [ warning sync.networkjob ]:	OCC::JsonJob(OCC::Account("myName@server.redacted.url"), "https://server.redacted.url/owncloud/owncloud/app/list", "GET", Original-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4", X-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4") Unsupported redirect on "https://server.redacted.url/owncloud/owncloud/app/list" to "https://server.redacted.url/owncloud/apps/files/"
23-11-21 19:53:20:136 [ debug sync.networkjob.jobqueue ]	[ OCC::JobQueue::retry ]:	Retry queued OCC::JsonJob(OCC::Account("myName@server.redacted.url"), "https://server.redacted.url/owncloud/owncloud/app/list", "GET", Original-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4", X-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4")
```

(Only posting a small snippet here since this log is too big to thoroughly sanitize. Holler if I missed anything important, but given the other comments, it looks like you've got this covered.)

**Comment by TheOneRing** [comment]
> > @Synthlight could you please run with the http logger? I'm interested in the response to https://server.redacted.url/owncloud/owncloud/app/list
> > It looks like the request to https://server.redacted.url/owncloud/owncloud/app/list fails and this triggers the account to revalidate its connection state. Possibly sending it into a loop.
> 
> It gets a 302/redirect.
> 
> I also see `Unsupported redirect on "https://server.redacted.url/owncloud/owncloud/app/list" to "https://server.redacted.url/owncloud/apps/files/"` in the log.
> 
> ```
> 23-11-21 19:53:20:136 [ info sync.httplogger ]:	"2191fb4e-d3a2-476a-9b60-0b536590c3c4: Response: GET 302 (Piplined,405ms) https://server.redacted.url/owncloud/owncloud/app/list Header: { Date: Wed, 22 Nov 2023 00:53:19 GMT, Server: Apache/2.4.56 (Debian), Strict-Transport-Security: max-age=15552000; includeSubDomains, X-Content-Type-Options: nosniff, X-XSS-Protection: 0, X-Robots-Tag: none, X-Frame-Options: SAMEORIGIN, X-Download-Options: noopen, X-Permitted-Cross-Domain-Policies: none, Expires: Thu, 19 Nov 1981 08:52:00 GMT, Cache-Control: no-store, no-cache, must-revalidate, Pragma: no-cache, Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *, Location: https://server.redacted.url/owncloud/apps/files/, Content-Length: 0, Content-Type: text/html; charset=UTF-8, Set-Cookie: oc_sessionPassphrase=__redacted__; expires=Wed, 22-Nov-2023 01:13:19 GMT; Max-Age=1200; path=/owncloud; secure; HttpOnly; SameSite=Strict, Keep-Alive: timeout=5, max=97, Connection: Keep-Alive, } Data: []"
> 23-11-21 19:53:20:136 [ warning gui.account.state ]:	checkConnectivity blocking: true
> 23-11-21 19:53:20:136 [ debug sync.networkjob.jobqueue ]	[ OCC::JobQueue::block ]:	block: 1 "myName@server.redacted.url"
> 23-11-21 19:53:20:136 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::checkServer ]:	Checking server and authentication
> 23-11-21 19:53:20:136 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::checkServer ]:	Trying to look up system proxy
> 23-11-21 19:53:20:136 [ warning sync.networkjob ]:	OCC::JsonJob(OCC::Account("myName@server.redacted.url"), "https://server.redacted.url/owncloud/owncloud/app/list", "GET", Original-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4", X-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4") Unsupported redirect on "https://server.redacted.url/owncloud/owncloud/app/list" to "https://server.redacted.url/owncloud/apps/files/"
> 23-11-21 19:53:20:136 [ debug sync.networkjob.jobqueue ]	[ OCC::JobQueue::retry ]:	Retry queued OCC::JsonJob(OCC::Account("myName@server.redacted.url"), "https://server.redacted.url/owncloud/owncloud/app/list", "GET", Original-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4", X-Request-ID: "2191fb4e-d3a2-476a-9b60-0b536590c3c4")
> ```
> 
> (Only posting a small snippet here since this log is too big to thoroughly sanitize. Holler if I missed anything important, but given the other comments, it looks like you've got this covered.)

See https://github.com/owncloud/client/issues/11382#issuecomment-1821162122 as a workaround.

**Comment by TheOneRing** [comment]
Fixed in https://github.com/owncloud/client/pull/11389 available in https://github.com/owncloud/client/releases/tag/v5.2.0-rc.1

**Comment by Synthlight** [comment]
Thanks for the fix! I can confirm v5.2.0-rc.1 solves my issue.
